lib/widgets/content.dart

```diff
@@ -571,26 +572,32 @@ class MessageImage extends StatelessWidget {
     final message = InheritedMessage.of(context);
 
     // TODO image hover animation
-    final src = node.srcUrl;
-
+    final srcUrl = node.srcUrl;
+    final thumbnailUrl = node.thumbnailUrl;
     final store = PerAccountStoreWidget.of(context);
-    final resolvedSrc = store.tryResolveUrl(src);
+    final resolvedSrcUrl = store.tryResolveUrl(srcUrl);
+    final resolvedThumbnailUrl = thumbnailUrl == null
+      ? null : store.tryResolveUrl(thumbnailUrl);
+
     // TODO if src fails to parse, show an explicit "broken image"
 
     return MessageMediaContainer(
-      onTap: resolvedSrc == null ? null : () { // TODO(log)
+      onTap: resolvedSrcUrl == null ? null : () { // TODO(log)
         Navigator.of(context).push(getLightboxRoute(
           context: context,
           message: message,
-          src: resolvedSrc,
+          src: resolvedSrcUrl,
+          thumbnailUrl: resolvedThumbnailUrl,
           mediaType: MediaType.image));
       },
-      child: resolvedSrc == null ? null : LightboxHero(
-        message: message,
-        src: resolvedSrc,
-        child: RealmContentNetworkImage(
-          resolvedSrc,
-          filterQuality: FilterQuality.medium)));
+      child: node.loading
+        ? const CupertinoActivityIndicator()
```

Comment:
Also forgot to flag this earlier, the reason for not using the `loader-{black,white}.svg` asset here is that, currently animated icons can't be included in ZulipIcons font (don't know if animated fonts are even possible), I tried and it just displayed a single (first) frame.

So, for now used the `CupertinoActivityIndicator` widget which conveniently is similar to the svg.

Reply:
Handy that that's there!

Yeah, I suspect animated fonts are not possible. If we were determined to use the icons from those SVGs, we could find a way (if nothing else, we could rasterize them to high-enough-resolution PNGs). But using this widget instead is a lot simpler and SGTM.

Would you post a screenshot and a video showing what this placeholder looks like? That doesn't require any fiddling on the server — the quick hack I used just now is:
```diff
--- lib/model/content.dart
+++ lib/model/content.dart
@@ -1064,7 +1064,7 @@ class _ZulipContentParser {
     if (href == null) {
       return UnimplementedBlockContentNode(htmlNode: divElement);
     }
-    if (imgElement.className == 'image-loading-placeholder') {
+    if (true || imgElement.className == 'image-loading-placeholder') {
       return ImageNode(
         srcUrl: href,
         thumbnailUrl: null,
```

Reply:
Added a video showing placeholders in the PR description.